examples/thread_duel/main.c

```diff
@@ -0,0 +1,203 @@
+        bad_wait(work * WORK_SCALE);
+        w += work;
+        nice_wait(rest * WORK_SCALE);
+    }
+}
+
+void * thread_bad(void * d)
+{
+    xtimer_usleep(200 * 1000);  /*always be nice at start*/
+#ifdef DEVELHELP
+    const char *name = thread_get_active()->name;
+#else
+    int16_t pid = thread_getpid();
+#endif
+
+    uint32_t w = 0;
+    uint32_t work = *((uint32_t *) d);
+    if (work > 10) {
+        work = 5;
+    }
+    uint32_t rest = (10 - work);
+    uint32_t step = 0;
+
+    /*work some time yield rest blocking*/
+    for (;;) {
+        if (w - step >= PRINT_STEPS) {
+#ifdef DEVELHELP
+            printf("%s: %" PRIu32 ", %" PRIu32 "\n", name, w, work);
+#else
+            printf("T-Pid %i:%" PRIu32 ", %" PRIu32 "\n", pid, w, work);
+#endif
+            step = w;
+        }
+        bad_wait(work * WORK_SCALE);
+        w += work;
+        bad_wait(rest * WORK_SCALE);
+    }
+}
+
+void * thread_restless_yielding(void * d)
+{
+    xtimer_usleep(200 * 1000);  /*always be nice at start*/
+#ifdef DEVELHELP
+    const char *name = thread_get_active()->name;
+#else
+    int16_t pid = thread_getpid();
+#endif
+
+    uint32_t w = 0;
+    uint32_t work = *((uint32_t *) d);
+    if (work > 10) {
+        work = 5;
+    }
+    /*uint32_t rest = (10 - work);*/
+    uint32_t step = 0;
+
+    /*work for some time and yield*/
+    for (;;) {
+        if (w - step >= PRINT_STEPS) {
+#ifdef DEVELHELP
+            printf("%s: %" PRIu32 ", %" PRIu32 "\n", name, w, work);
+#else
+            printf("T-Pid %i:%" PRIu32 ", %" PRIu32 "\n", pid, w, work);
+#endif
+            step = w;
+        }
+        bad_wait(work * WORK_SCALE);
+        w += work;
+        thread_yield();
+    }
+}
+
+void * thread_restless(void * d)
+{
+    xtimer_usleep(200 * 1000);  /*always be nice at start*/
+#ifdef DEVELHELP
+    const char * name = thread_get_active()->name;
+#else
+    int16_t pid = thread_getpid();
+#endif
+
+    uint32_t w = 0;
+    uint32_t work = *((uint32_t *) d);
+    if (work > 10) {
+        work = 5;
+    }
+    /*uint32_t rest = (10 - work);*/
+    uint32_t step = 0;
+
+    /*continue working*/
+    for (;;) {
+        if (w - step >= PRINT_STEPS) {
+#ifdef DEVELHELP
+            printf("%s: %" PRIu32 ", %" PRIu32 "\n", name, w, work);
+#else
+            printf("T-Pid %i:%" PRIu32 ", %" PRIu32 "\n", pid, w, work);
+#endif
+            step = w;
+        }
+        bad_wait(work * WORK_SCALE);
+        w += work;
+    }
+}
+
+int main(void)
+{
+    {
+        static char stack[THREAD_STACKSIZE_DEFAULT];
+        static uint32_t workness = 3;   /* 0-10 workness*/
+        thread_create(stack, sizeof(stack), 7, THREAD_CREATE_STACKTEST,
+                      thread_restless, &workness, "T1");
+    }
+    {
+        static char stack[THREAD_STACKSIZE_DEFAULT];
+        static uint32_t workness = 5;   /* 0-10 workness*/
+        thread_create(stack, sizeof(stack), 7, THREAD_CREATE_STACKTEST,
+                      thread_restless, &workness, "T2");
```

Comment:
Can we use MACROS for how thsese are assigned and warp them in:

```
#ifndef   FIRST_THREAD_HANDLER
#define  FIRST_THREAD_HANDLER  thread_restless
#endif 
```

That way I could change the application behaviour with CFLAGS and overrriding the behaviour.